@here/harp-materials/lib/ShaderChunks/ExtrusionChunks.ts

```diff
@@ -4,15 +4,27 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import { ExtrusionFeatureDefs } from "../MapMeshMaterialsDefs";
+
+const MIN_BUILDING_HEIGHT_SQUARED =
+    ExtrusionFeatureDefs.MIN_BUILDING_HEIGHT * ExtrusionFeatureDefs.MIN_BUILDING_HEIGHT;
+
 export default {
     extrusion_pars_vertex: `
 // Extrusion axis (xyz: vector, w: factor).
 attribute vec4 extrusionAxis;
 uniform float extrusionRatio;
 varying vec4 vExtrusionAxis;
+varying float vExtrusionRatio;
+
 `,
     extrusion_vertex: `
-transformed = transformed + extrusionAxis.xyz * (extrusionRatio - 1.0);
+// Cancel extrusionRatio (meaning, force to 1) if extrusionAxisLen < MIN_BUILDING_HEIGHT.
+const float MIN_BUILDING_HEIGHT_SQUARED = ${MIN_BUILDING_HEIGHT_SQUARED};
```

Comment:
I am a bit worried if min building height shouldn't be function of camera distance:
`const minHeight = calcMinHeight(modelViewMatrix)`
Could you check how it behaves in different geo projection setup with camera place far above the ground.

Reply:
For normal building - root cause of issue, current solution should be ok, as they are visible on levels where current value of MIN_BUILDING_HEIGHT is significant, let's deliver this one and implement fully robust one later if really needed.

Reply:
Agreed!